test/uk/gov/hmrc/agentsubscription/model/ValidatorSpec.scala

```diff
@@ -188,29 +338,226 @@ class ValidatorSpec extends UnitSpec {
       }
 
       "there is whitespace" in {
-        validateAddress("     ") shouldBe whitespaceValidationError
+        validateName("     ") shouldBe whitespaceValidationError
       }
 
       "string is empty but field is present" in {
-        validateAddress("") shouldBe whitespaceValidationError
+        validateName("") shouldBe whitespaceValidationError
       }
     }
   }
 
-  private def validatePostcode(input: String) = {
-    postcodeValidation.reads(JsString(input))
-  }
+  "overseasNameValidation" should {
+    def validateOSName(name: String) = overseasNameValidation.reads(JsString(name))
+    def nameValidationError = JsError(ValidationError("error.name.invalid"))
+
+    "accept name" when {
+      "a valid name is provided" in {
+        validateOSName("Number 1 Foo/Acme Test-Agency, Ltd.") shouldBe JsSuccess("Number 1 Foo/Acme Test-Agency, Ltd.")
+      }
+
+      "contains uppercase letters" in {
+        validateOSName("ABCDEFGHIJKLMNOPQRSTUVWXYZ") shouldBe JsSuccess("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+      }
 
-  private def validatePhoneNumber(number: String) = {
-    telephoneNumberValidation.reads(JsString(number))
+      "contains lowercase letters" in {
+        validateOSName("abcdefghijklmnopqrstuvwxyz") shouldBe JsSuccess("abcdefghijklmnopqrstuvwxyz")
+      }
+
+      "contains numbers" in {
+        validateOSName("0123456789") shouldBe JsSuccess("0123456789")
+      }
+
+      "contains spaces" in {
+        validateOSName(" A B C ") shouldBe JsSuccess(" A B C ")
+      }
+
+      "contains hyphens -" in {
+        validateOSName("-") shouldBe JsSuccess("-")
+      }
+
+      "contains commas ," in {
+        validateOSName(",") shouldBe JsSuccess(",")
+      }
+
+      "contains full stops ." in {
+        validateOSName(".") shouldBe JsSuccess(".")
+      }
+
+      "contains forward slashes /" in {
+        validateOSName("/") shouldBe JsSuccess("/")
+      }
+
+      "1 or more characters" in {
+        validateOSName("1") shouldBe JsSuccess("1")
+      }
+
+      "40 or fewer characters" in {
+        validateOSName("1234567890123456789012345678901234567890") shouldBe JsSuccess("1234567890123456789012345678901234567890")
+      }
+    }
+
+    "reject name" when {
+      "contains ampersands &" in {
+        validateOSName("&") shouldBe nameValidationError
+      }
+
+      "contains single quotes '" in {
+        validateOSName("'") shouldBe nameValidationError
+      }
+
+      "contains backslashes \\" in {
+        validateOSName("\\") shouldBe nameValidationError
+      }
+
+      "contains other invalid symbols" in {
+        validateOSName("Agency;Co") shouldBe nameValidationError
+        validateOSName("#1 Agency Worldwide") shouldBe nameValidationError
+        validateOSName("|Agency|") shouldBe nameValidationError
+      }
+
+      "more than 40 characters" in {
+        validateOSName("12345678901234567890123456789012345678901") shouldBe nameValidationError
```

Comment:
it be really nice to have .lenght or size checks for long inputs, just like you have done with 133 check

Reply:
Done